crates/router/src/core/payments/operations/payment_create.rs

```diff
@@ -544,7 +546,28 @@ impl PaymentCreate {
             .transpose()
             .change_context(errors::ApiErrorResponse::InternalServerError)
             .attach_printable("Encoding Metadata to value failed")?;
-
+        let order_details_metadata_req =
+            request.clone().metadata.and_then(|meta| meta.order_details);
+        if request
+            .clone()
+            .order_details
+            .zip(order_details_metadata_req)
+            .is_some()
+        {
+            Err(errors::ApiErrorResponse::NotSupported { message: "order_details cannot be present both inside and outside metadata in payments request".to_string() })?
+        }
+        let order_details_outside_value = request.clone().order_details.map(|o| {
+            o.iter()
+                .map(|order| {
+                    let encoded_order = Encode::<OrderDetailsWithAmount>::encode_to_value(&order)
+                        .change_context(errors::ApiErrorResponse::NotSupported {
+                            message: "failed while encoding order_details".to_string(),
+                        })
+                        .unwrap_or_default();
```

Comment:
If any errors occurs, we need to pass default?

If so why we need to change_context?

If error occurs, how we know about that error.